t/plugin/public-api.t

```diff
@@ -0,0 +1,219 @@
+                        "plugins": {
+                            "jwt-auth": {
+                                "key": "user-key",
+                                "algorithm": "HS256"
+                            }
+                        }
+                    }]]
+                },
+                {
+                    uri = "/apisix/admin/routes/custom-jwt-sign",
+                    data = [[{
+                        "plugins": {
+                            "public-api": {
+                                "uri": "/apisix/plugin/jwt/sign"
+                            },
+                            "serverless-pre-function": {
+                                "phase": "rewrite",
+                                "functions": ["return function(conf, ctx) require(\"apisix.core\").log.warn(\"custom-jwt-sign was triggered\"); end"]
+                            }
+                        },
+                        "uri": "/gen_token"
+                    }]],
+                },
+                {
+                    uri = "/apisix/admin/routes/direct-wolf-rbac-userinfo",
+                    data = [[{
+                        "plugins": {
+                            "public-api": {},
+                            "serverless-pre-function": {
+                                "phase": "rewrite",
+                                "functions": ["return function(conf, ctx) require(\"apisix.core\").log.warn(\"direct-wolf-rbac-userinfo was triggered\"); end"]
+                            }
+                        },
+                        "uri": "/apisix/plugin/wolf-rbac/user_info"
+                    }]],
+                },
+            }
+
+            local t = require("lib.test_admin").test
+
+            for _, data in ipairs(datas) do
+                local code, body = t(data.uri, ngx.HTTP_PUT, data.data)
+                ngx.say(code..body)
+            end
+        }
+    }
+--- response_body eval
+"201passed\n" x 3
+
+
+
+=== TEST 3: hit route (custom-jwt-sign)
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+
+            local code, body, jwt = t("/gen_token?key=user-key", ngx.HTTP_GET, "", nil, {apikey = "testkey"})
+            if code >= 300 then
+                ngx.status = code
+            end
+
+            local header = string.sub(jwt, 1, 36)
+
+            if header == "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9" or
+               header == "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9" then
+                ngx.say("passed")
+                return
+            end
+
+            ngx.say("failed")
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 4: hit route (direct-wolf-rbac-userinfo)
+--- request
+GET /apisix/plugin/wolf-rbac/user_info
+--- error_code: 401
+--- error_log
+direct-wolf-rbac-userinfo was triggered
+
+
+
+=== TEST 5: missing route
+--- request
+GET /apisix/plugin/balalbala
+--- error_code: 404
```

Comment:
Let's add more test that the uri in public API plugin matches nothing.

Reply:
got it, adding

Reply:
added